routers/web/repo/actions/view.go

```diff
@@ -581,6 +581,33 @@ func Approve(ctx *context_module.Context) {
 	ctx.JSON(http.StatusOK, struct{}{})
 }
 
+func Delete(ctx *context_module.Context) {
+	runIndex := getRunIndex(ctx)
+	repoID := ctx.Repo.Repository.ID
+
+	run, err := actions_model.GetRunByIndex(ctx, repoID, runIndex)
+	if err != nil {
+		if errors.Is(err, util.ErrNotExist) {
+			ctx.HTTPError(http.StatusNotFound, err.Error())
+			return
+		}
+		ctx.HTTPError(http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	if !run.Status.IsDone() {
+		ctx.JSONError(ctx.Tr("actions.runs.not_done"))
+		return
+	}
+
+	if err := actions_service.DeleteRun(ctx, run); err != nil {
+		ctx.HTTPError(http.StatusInternalServerError, err.Error())
```

Comment:
No, do not expose internal err message to end users, it may contain sensitive information and lead to security problems.

And avoid using `ctx.HTTPError(http.StatusInternalServerError`, take a look at other code.

Reply:
> There are lots of such inconsistency you mention in the whole Gitea codebase. 🤷‍♂️

Feel free to point out these abuses and we need to fix them .........